move_group/src/default_capabilities/move_action_capability.cpp

```diff
@@ -61,8 +61,26 @@ void move_group::MoveGroupMoveAction::initialize()
 void move_group::MoveGroupMoveAction::executeMoveCallback(const moveit_msgs::MoveGroupGoalConstPtr& goal)
 {
   setMoveState(PLANNING);
-  context_->planning_scene_monitor_->updateFrameTransforms();
 
+  const ros::Time now = ros::Time::now();
+  for (int i = 0; ; ++i)
+  {
+    ros::Duration age = now - context_->planning_scene_monitor_->getLastRobotStateUpdateTime();
+    if (age < ros::Duration(0))
+      break;
+
+    if (i >= 200)
+    {
+      ROS_WARN("Failed to get current robot state!");
+      move_action_server_->setAborted();
+      setMoveState(IDLE);
+      return;
+    }
+
+    ros::Duration(0.001).sleep();
+  }
```

Comment:
I believe this whole block should only be executed if `planning_scene::isEmpty(goal->request.start_state)`.
If you just want to plan with a complete state, there is no need to wait for a state update.
On the other hand, if the start_state is empty that means we should plan with the current state, so we should wait for the next update as you proposed @andreaskoepf .